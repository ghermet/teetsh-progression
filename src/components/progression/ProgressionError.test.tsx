/** Generated with CLAUDE CODE */

import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { ProgressionError } from './ProgressionError';

describe('ProgressionError', () => {
  const mockError = new Error('Test error message');
  const mockReset = vi.fn();

  it('renders the error message correctly', () => {
    render(<ProgressionError error={mockError} reset={mockReset} />);

    expect(screen.getByText('Erreur de chargement')).toBeInTheDocument();
    expect(screen.getByText('Test error message')).toBeInTheDocument();
  });

  it('calls the reset function when the button is clicked', () => {
    render(<ProgressionError error={mockError} reset={mockReset} />);

    const button = screen.getByRole('button', { name: /RafraÃ®chir la page/i });
    fireEvent.click(button);

    expect(mockReset).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { container } = render(
      <ProgressionError error={mockError} reset={mockReset} />
    );
    expect(container).toMatchSnapshot();
  });
});
