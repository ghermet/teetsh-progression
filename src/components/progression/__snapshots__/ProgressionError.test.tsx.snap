// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressionError > matches snapshot 1`] = `
<div>
  <div
    class="w-full max-w-2xl mx-auto px-4 sm:px-6 py-8 sm:py-12 min-h-screen flex items-start justify-center"
  >
    <div
      class="bg-red-50 border border-red-200 rounded-lg sm:rounded-xl p-6 sm:p-8 w-full"
    >
      <h1
        class="text-2xl sm:text-3xl font-semibold text-red-800 mb-3 sm:mb-4"
      >
        Erreur de chargement
      </h1>
      <p
        class="text-red-600 mb-6 sm:mb-8 text-sm sm:text-base"
      >
        Test error message
      </p>
      <button
        class="w-full sm:w-auto px-6 py-3 font-semibold text-white bg-red-500 rounded-md hover:bg-red-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 transition-colors touch-manipulation"
      >
        Rafra√Æchir la page
      </button>
    </div>
  </div>
</div>
`;
